src/monster.cpp

```diff
@@ -146,54 +146,41 @@ void monster::poly(mtype *t)
 
 bool monster::can_upgrade() const
 {
-    // If we don't upgrade
+    // Turned off means turned off
+    if (!ACTIVE_WORLD_OPTIONS["MONSTER_UPGRADE_FACTOR"]) {
```

Comment:
> It was better with an explicit comparison rather than implicit cast to bool.

Agreed.

> EDIT: Not sure if performance is a concern here, but if it is, it should be moved near the end of the function.

Or maybe somewhere else altogether. If it's 0/false then the whole upgrade stuff is disabled (which is why I check this first, why bother with the rest if it's disabled).